modules/ethereum/src/benchmarking.rs

```diff
@@ -55,6 +60,163 @@ benchmarks! {
 
 	}: import_unsigned_header(RawOrigin::None, header, None)
 	verify {
-		assert_eq!(BridgeStorage::<T>::new().best_block().0.number, 1);
+		let storage = BridgeStorage::<T>::new();
+		assert_eq!(storage.best_block().0.number, 1);
+		assert_eq!(storage.finalized_block().number, 0);
+	}
+
+	// Our goal with this bench is to try and see the effect that finalizing difference ranges of
+	// blocks has on our import time. As such we need to make sure that we keep the number of
+	// validators fixed while changing the number blocks finalized (the complixity parameter) by
+	// importing the last header.
+	import_unsigned_finality {
+		// Our complexity parameter, n, will represent the number of blocks imported before
+		// finalization.
+		//
+		// For two validators this is only going to work for even numbers...
+		let n in 4..10;
+
+		// This should remain fixed for the bench.
+		let num_validators: u32 = 2;
+
+		let mut storage = BridgeStorage::<T>::new();
+
+		// Initialize storage with some initial header
+		let initial_header = build_genesis_header(&validator(0));
+		let initial_header_hash = initial_header.compute_hash();
+		let initial_difficulty = initial_header.difficulty;
+		let initial_validators = validators_addresses(num_validators as usize);
+
+		initialize_storage::<T>(
+			&initial_header,
+			initial_difficulty,
+			&initial_validators,
+		);
+
+		let mut headers = Vec::new();
+		let mut ancestry: Vec<FinalityAncestor<Option<Address>>> = Vec::new(); // Is this needed?
+		let mut parent = initial_header.clone();
+
+		// Q: Is this a sketchy way of "delaying" finality, or should I be manually editing the
```

Comment:
Regular import of this header should fail because of 'wrong step', but since you're inserting header without actual verification, it is ok.